src/System.Net.Http/tests/FunctionalTests/SyncBlockingContent.cs

```diff
@@ -0,0 +1,26 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.IO;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace System.Net.Http.Functional.Tests
+{
+    public sealed class SyncBlockingContent : StringContent
+    {
+        byte[] _content;
+
+        public SyncBlockingContent(string content) : base(content)
+        {
+            _content = Encoding.UTF8.GetBytes(content);
+        }
+
+        protected override Task SerializeToStreamAsync(Stream stream, TransportContext context)
+        {
+            stream.Write(_content, 0, _content.Length);
```

Comment:
I think this test is a bit different from the WCF scenario.  This is a valid test case, but I believe their situation is a bit different, in that the synchronous Write is happening in response to data being read from stream (since they have a producer-consumer stream).  I mention this because I believe I know what the issue is with CurlHandler, which would correctly handle that WCF case but which hangs if SerializeToStreamAsync itself synchronously writes to the stream.  We probably need both test cases.